### PR TITLE
[*] Apimap - Allow users to use field named `type`

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Pull Request checklist:

- [X] Write an explicit title for the Pull Request
- [X] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [ ] No automatic tests failures
- [ ] Test manually the implemented changes
- [X] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by VincentMolinie** [comment]
[(?) P3 - field name "type" is not supported in Forest](https://trello.com/c/PVPIhvAi/2801-p3-field-name-type-is-not-supported-in-forest)

File: app/services/forest_liana/schema_adapter.rb [hunk 1]
```diff
@@ -336,7 +336,8 @@ def add_enum_values_if_is_sti_model(column_schema, column)
 
     def sti_column?(column)
       (@model.inheritance_column &&
-       column.name == @model.inheritance_column) || column.name == 'type'
+        column.name == @model.inheritance_column) ||
+        (!@model.inheritance_column && column.name == 'type')
```
Comment (arnaudbesnier) [inline app/services/forest_liana/schema_adapter.rb]: STI parent that does not have children models should actually have their `inheritance_column` displayed even if it is the default `type` column.

Comment (VincentMolinie) [reply app/services/forest_liana/schema_adapter.rb]: That is actually pretty smart, I'm gonna check your proposition

**Comment by VincentMolinie** [comment]
Should I remove this PR ?

**Comment by arnaudbesnier** [comment]
If the other one makes a better job, the answer is yes.

**Pull request closed without merging**